### DISCUSSION TITLE
ElasticSearch is not using the most recent scroll_id

### DISCUSSION THREAD

**Discussion by cdegroc** [JanusGraph/janusgraph#2580] [Q&A] [answered]
[ElasticSearch documentation](https://www.elastic.co/guide/en/elasticsearch/reference/current/paginate-search-results.html#scroll-search-results) has the following statement about Scroll Searches:
> Important. The initial search request and each subsequent scroll request each return a _scroll_id. While the _scroll_id may change between requests, it doesn’t always change — in any case, only the most recently received _scroll_id should be used.

If I'm not mistaken, I don't think the latest scroll_id is used in JanusGraph.

In `ElasticSearchScroll`:
- `this.scrollId` is set to the `_scroll_id` from the initial request's response: https://github.com/JanusGraph/janusgraph/blob/master/janusgraph-es/src/main/java/org/janusgraph/diskstorage/es/ElasticSearchScroll.java#L41
- new scroll requests are made using `this.scrollId`: https://github.com/JanusGraph/janusgraph/blob/master/janusgraph-es/src/main/java/org/janusgraph/diskstorage/es/ElasticSearchScroll.java#L65
- `this.scrollId` is not updated when a new response is received: https://github.com/JanusGraph/janusgraph/blob/master/janusgraph-es/src/main/java/org/janusgraph/diskstorage/es/ElasticSearchScroll.java#L46-L54

Can someone please confirm this is an issue? Then I can submit a PR if needed.

**Comment by li-boxuan** [comment, marked as answer]
I think you are correct. This looks like a bug to me. Although scrollId does not change most of the time (https://discuss.elastic.co/t/scroll-id-is-not-changing-while-querying/106202/3), we should definitely follow what the official documentation says. It would be great if you could submit a PR!
